Grammar/python.gram

```diff
@@ -858,6 +858,7 @@ invalid_expression:
     # Soft keywords need to also be ignored because they can be parsed as NAME NAME
    | !(NAME STRING | SOFT_KEYWORD) a=disjunction b=expression_without_invalid {
         RAISE_SYNTAX_ERROR_KNOWN_RANGE(a, b, "invalid syntax. Perhaps you forgot a comma?") }
+   | a=disjunction 'if' b=disjunction NEWLINE { RAISE_SYNTAX_ERROR("invalid syntax. Conditional expression expected an 'else' here.") }
```

Comment:
Using NEWLINE here is unfortunately not correct as is not respecting the nesting appropriately. For instance, one could have:

```
[X, Y, 4 if 5, W]
```

And the error won't trigger.

Reply:
I will suggest to use this diff instead:

```diff
diff --git a/Grammar/python.gram b/Grammar/python.gram
index 278ddfb2d1..f55e83fa57 100644
--- a/Grammar/python.gram
+++ b/Grammar/python.gram
@@ -1083,6 +1083,7 @@ invalid_expression:
     # Soft keywords need to also be ignored because they can be parsed as NAME NAME
    | !(NAME STRING | SOFT_KEYWORD) a=disjunction b=expression_without_invalid {
         RAISE_SYNTAX_ERROR_KNOWN_RANGE(a, b, "invalid syntax. Perhaps you forgot a comma?") }
+   | a=disjunction 'if' b=disjunction !'else' { RAISE_SYNTAX_ERROR_KNOWN_RANGE(a, b, "expected 'else' after 'if' expression") }

 invalid_named_expression:
     | a=expression ':=' expression {
```

You can then add some tests for the example where the expression is nested. Such as:

```
>>> y = [1,2,x if 3,4]
  File "<stdin>", line 1
    y = [1,2,x if 3,4]
             ^^^^^^
SyntaxError: expected 'else' after 'if' expression
```

Reply:
Thanks for flagging this @pablogsal , I've pushed your suggestion!

ps: I didn't know one could do `!'else'`

Reply:
I'm seeing tests failing but I'm not sure why. It works testing manually in the console and running the tests work.

I did `make regen-all` and `make -j8 -s`.

I'm testing on Ubuntu.

```bash
$ cat /etc/os-release
NAME="Ubuntu"
VERSION="18.04.5 LTS (Bionic Beaver)"
```

Reply:
I have made the requested changes; please review again